peekingduck/pipeline/nodes/model/yolov4/yolo_model.py

```diff
@@ -36,47 +36,52 @@ def __init__(self, config: Dict[str, Any]) -> None:
         if not 0 <= config["yolo_score_threshold"] <= 1:
             raise ValueError("yolo_score_threshold must be in [0, 1]")
 
+        # check for yolo weights, if none then download into weights folder
         weights_dir, model_dir = finder.find_paths(
             config["root"], config["weights"], config["weights_parent_dir"]
         )
-
-        # check for yolo weights, if none then download into weights folder
         if not checker.has_weights(weights_dir, model_dir):
             self.logger.info("---no weights detected. proceeding to download...---")
             downloader.download_weights(weights_dir, config["weights"]["blob_file"])
             self.logger.info(f"---weights downloaded to {weights_dir}.---")
 
         classes_path = model_dir / config["weights"]["classes_file"]
-
         with open(classes_path) as infile:
             self.class_names = [c.strip() for c in infile.readlines()]
+
+        self.detector = Detector(config, model_dir, self.class_names)
         self.detect_ids = config["detect_ids"]
 
-        self.detector = Detector(config, model_dir)
+    @property
+    def detect_ids(self) -> List[int]:
+        """The list of selected object category IDs."""
+        return self._detect_ids
+
+    @detect_ids.setter
+    def detect_ids(self, ids: List[int]) -> None:
+        if not isinstance(ids, list):
+            raise TypeError("detect_ids has to be a list")
+        if not ids:
+            self.logger.info("Detecting all Yolo classes")
+        self._detect_ids = ids
 
     def predict(
-        self, frame: np.ndarray
+        self, image: np.ndarray
     ) -> Tuple[List[np.ndarray], List[str], List[float]]:
         """predict the bbox from frame
 
+        Args:
+            image (np.ndarray): Input image frame.
+
         Returns:
             object_bboxes (List[Numpy Array]): list of bboxes detected
             object_labels (List[str]): list of string labels of the
                 object detected for the corresponding bbox
             object_scores (List(float)): list of confidence scores of the
                 object detected for the corresponding bbox
         """
-        assert isinstance(frame, np.ndarray)
+        if not isinstance(image, np.ndarray):
+            raise TypeError("image must be a np.ndarray")
 
         # return bboxes, object_bboxes, object_labels, object_scores
```

Comment:
Consider either updating this comment like you've done in `efficientdet_model.py` or removing it

Reply:
Done: thanks for catching this, missed it.